[SYCL][Graph] Release host sync for multiple command-buffer submissions (#345)
reble/llvm: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

Reviewer: Couple of other spec sections that need touched:

* In the "Executable Graph Update" future work section, could you document that we will allow concurrent submissions after a graph has been updated.
* In the "Issues" section could you remove the sub-section for "Simultaneous Graph Submission"

handler::ext_oneapi_graph(command_graph<graph_state::executable>& graph)
----

|Invokes the execution of a graph. Only one instance of `graph` will
execute at any time. If `graph` is submitted multiple times, dependencies
are automatically added by the runtime to prevent concurrent executions of 
an identical graph. Note that synchronizations must be handled in the host 

> for level-zero backend, which may involve extra latency for subsequent 

Reviewer: I'd move this last level zero specific sentence to the section at the end on "Non-implemented features and known issues"

handler::ext_oneapi_graph(command_graph<graph_state::executable>& graph)
----

> |Invokes the execution of a graph. Only one instance of `graph` may be 
> executing, at any time. If a `graph` is submitted multiple times dependencies
> are automatically added to prevent multiple simultaneous executions of 
> an identical graph. 

Reviewer: "may be executing" isn't strong enough wording now we're enforcing this.
```suggestion
|Invokes the execution of a graph. Only one instance of `graph` will
execute at any time. If `graph` is submitted multiple times, dependencies
are automatically added by the runtime to prevent concurrent executions of 
an identical graph. 
```

Concurrent graph execution can be achieved by finalizing a graph in 

> modifiable state into multiple graphs in executable state.

Reviewer: > Concurrent graph execution can be achieved by finalizing a graph in 
modifiable state into multiple graphs in executable state.

Can we remove this sentence, I think it's flawed advice. If a graph is non-trivial then the multiple executable graphs finalized from the same modifiable graph will have race conditions if executing concurrently.

Reviewer 2: agree to remove this sentence, it also aligns with https://github.com/intel/llvm/pull/11675
